Formula/libglib.rb

```diff
@@ -0,0 +1,109 @@
+class Libglib < Formula
+  desc "Core application library for C"
+  homepage "https://developer.gnome.org/glib/"
+  url "https://download.gnome.org/sources/glib/2.72/glib-2.72.2.tar.xz"
+  sha256 "78d599a133dba7fe2036dfa8db8fb6131ab9642783fc9578b07a20995252d2de"
+  license "LGPL-2.1-or-later"
+
+  livecheck do
+    formula "glib"
+  end
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
+  depends_on "gettext"
+  depends_on "pcre"
+
+  uses_from_macos "libffi"
+
+  on_linux do
+    depends_on "util-linux"
+  end
+
+  def install
+    # TODO: This is a workaround for `brew audit --new-formula`.
+    #       Use `patch` rather than `inreplace` (see also `glib`).
+    # replace several hardcoded paths with homebrew counterparts
+    inreplace "gio/xdgmime/xdgmime.c",
+              'xdg_data_dirs = "/usr/local/share/:/usr/share/";',
+              "xdg_data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/share/\";"
+    inreplace "glib/gutils.c",
+              'data_dirs = "/usr/local/share/:/usr/share/";',
+              "data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/share/\";"
```

Comment:
I don't think we should remove something here, especially since it's a path.

Reply:
These patches are grabbed from `glib.rb`.

Reply:
I think that patch is wrong too, please make an issue upstream asking to make that configurable.

Reply:
There is a bug report at <https://bugzilla.gnome.org/show_bug.cgi?id=673135>. In [comment 2](https://bugzilla.gnome.org/show_bug.cgi?id=673135#c2):

> Dan Winship 2012-03-30 12:26:48 UTC
> "/usr/local/share:/usr/share" is specified by the XDG directory spec (http://standards.freedesktop.org/basedir-spec/latest/ar01s03.html), meaning non-glib-based software will make the same assumption. If you install packages elsewhere, you have to set XDG_DATA_DIRS in the default environment to override the default definition. (Just like you have to update PATH in the default environment for people to be able to run the binaries in the first place.)

They hardcode these paths on purpose. Users need to set `XDG_DATA_DIRS` at runtime rather than at compile time.

I cannot find any related issue nor PR on GLib's official GitLab repo.